src/ClientData/include/ClientData/CallstackData.h

```diff
@@ -83,9 +83,32 @@ class CallstackData {
   template <typename Action>
   void ForEachCallstackEventInTimeRangeDiscretized(uint64_t min_timestamp, uint64_t max_timestamp,
                                                    uint32_t resolution, Action&& action) const {
-    for (const auto& [tid, unused_events] : callstack_events_by_tid_) {
-      ForEachCallstackEventOfTidInTimeRangeDiscretized(tid, min_timestamp, max_timestamp,
-                                                       resolution, action);
+    auto get_next_callstack = [&](uint64_t timestamp) -> CallstackEvent {
+      constexpr uint32_t kFakeId = -1;
+      CallstackEvent next_callstack{max_timestamp, kFakeId, kFakeId};
+      uint32_t current_pixel = GetPixelNumber(timestamp, resolution, min_timestamp, max_timestamp);
```

Comment:
The name of variable and the loop right after its declaration make me wonder if its value changes. And it seems like it doesn't. I suggest `const` here.

Reply:
... and as we const one variable, also const all the other constable variables.